app/abuser_utils.py

```diff
@@ -0,0 +1,272 @@
+        aad_bytes = constants.AEAD_AAD_DATA.encode("utf-8")
+        encrypted_bundle_data = nonce_bundle + aesgcm_bundle_enc.encrypt(
+            nonce_bundle, bundle_json_bytes, aad_bytes
+        )
+        abuser_data_entry = AbuserData(
+            user_id=user.id, encrypted_bundle=encrypted_bundle_data
+        )
+
+        Session.add(abuser_data_entry)
+        Session.flush()
+
+        blob_id = abuser_data_entry.id
+        all_identifiers_raw = (
+            [primary_email]
+            + [a.email for a in aliases if a.email]
+            + [m.email for m in mailboxes if m.email]
+        )
+        seen_normalized_identifiers = set()
+        mac_key_bytes = config.MAC_KEY
+        master_key_bytes = config.MASTER_ENC_KEY
+
+        for raw_identifier_address in all_identifiers_raw:
+            if not raw_identifier_address:
+                continue
+
+            normalized_identifier = raw_identifier_address.lower()
+
+            if normalized_identifier in seen_normalized_identifiers:
+                continue
+
+            seen_normalized_identifiers.add(normalized_identifier)
+            identifier_hmac = hmac.new(
+                mac_key_bytes, normalized_identifier.encode("utf-8"), sha256
+            ).hexdigest()
+
+            k_identifier_derived = _derive_key_for_identifier(
+                master_key_bytes, normalized_identifier
+            )
+            aesgcm_key_enc = AESGCM(k_identifier_derived)
+            nonce_key_encryption = secrets.token_bytes(12)
+            encrypted_k_bundle_for_this_identifier = (
+                nonce_key_encryption
+                + aesgcm_key_enc.encrypt(
+                    nonce_key_encryption, k_bundle_random, aad_bytes
+                )
+            )
+            abuser_lookup_entry = AbuserLookup(
+                hashed_address=identifier_hmac,
+                abuser_data_id=blob_id,
+                bundle_k=encrypted_k_bundle_for_this_identifier,
+            )
+
+            Session.add(abuser_lookup_entry)
+
+        Session.commit()
+        emit_abuser_audit_log(
+            user=actor,
+            action=AbuserAuditLogAction.MarkAbuser,
+            message=f"An user {user.id} was marked as abuser.",
+        )
+    except Exception:
+        Session.rollback()
+        LOG.exception("Error during archive_abusive_user")
+        raise
+
+
+def unmark_as_abusive_user(user_id: int, actor: User) -> None:
+    """
+    Fully remove abuser archive and lookup data for a given user_id.
+    This reverses the effects of archive_abusive_user().
+    """
+    abuser_data_entry = AbuserData.filter_by(user_id=user_id).first()
+
+    if not abuser_data_entry:
+        LOG.e("No abuser data found for user %s", user_id)
+        return
+
+    Session.delete(abuser_data_entry)
+    Session.commit()
+    emit_abuser_audit_log(
+        user=actor,
+        action=AbuserAuditLogAction.UnmarkAbuser,
+        message=f"An user {user_id} was unmarked as abuser.",
+    )
```

Comment:
```suggestion
def unmark_as_abusive_user(user_id: int, actor: User) -> None:
    """
    Fully remove abuser archive and lookup data for a given user_id.
    This reverses the effects of archive_abusive_user().
    """
    abuser_data_entry = AbuserData.filter_by(user_id=user_id).first()

    if not abuser_data_entry:
        LOG.e("No abuser data found for user %s", user_id)
        return

    Session.delete(abuser_data_entry)
    emit_abuser_audit_log(
        user=user_id,
        admin = actor
        action=AbuserAuditLogAction.UnmarkAbuser,
        message=f"An user {user_id} was unmarked as abuser.",
    )
    Session.commit()
```
We want to store in the audit log the user that was removed as abuser and the admin that did it